datastore/tasks.js

```diff
@@ -82,128 +83,112 @@ function addTask (description, callback) {
         value: false
       }
     ]
-  }, function (err) {
-    if (err) {
-      return callback(err);
-    }
-
-    var taskId = taskKey.path.pop();
-    console.log('Task %d created successfully.', taskId);
-    return callback(null, taskKey);
-  });
+  };
+
+  return datastore.save(entity)
+    .then(() => {
+      console.log(`Task ${taskKey.id} created successfully.`);
+      return taskKey;
+    });
 }
 // [END add_entity]
 
 // [START update_entity]
-function markDone (taskId, callback) {
-  var transaction = datastore.transaction();
-
-  transaction.run(function (err) {
-    if (err) {
-      return callback(err);
-    }
-
-    var taskKey = datastore.key([
-      'Task',
-      taskId
-    ]);
-
-    transaction.get(taskKey, function (err, task) {
-      if (err) {
-        // An error occurred while getting the task
-        return transaction.rollback(function (_err) {
-          return callback(err || _err);
-        });
-      }
-
-      task.data.done = true;
-
-      transaction.save(task);
-
-      // Commit the transaction
-      transaction.commit(function (err) {
-        if (err) {
-          return callback(err);
-        }
+function markDone (taskId) {
+  const transaction = datastore.transaction();
+  const taskKey = datastore.key([
+    'Task',
+    taskId
+  ]);
 
-        // The transaction completed successfully.
-        console.log('Task %d updated successfully.', taskId);
-        return callback(null);
+  return transaction.run()
+    .then(() => transaction.get(taskKey))
+    .then((results) => {
+      const task = results[0];
+      task.done = true;
+      transaction.save({
+        key: taskKey,
```

Comment:
Nit: can this be compressed by using ES6 and renaming variables? e.g. `key: taskKey` ==> `key`